datetime/datetime.go

```diff
@@ -0,0 +1,129 @@
+/**
+ * Datetime MessagePack serialization schema is an MP_EXT extension, which
+ * creates container of 8 or 16 bytes long payload.
+ *
+ * +---------+--------+===============+-------------------------------+
+ * |0xd7/0xd8|type (4)| seconds (8b)  | nsec; tzoffset; tzindex; (8b) |
+ * +---------+--------+===============+-------------------------------+
+ *
+ * MessagePack data encoded using fixext8 (0xd7) or fixext16 (0xd8), and may
+ * contain:
+ *
+ * - [required] seconds parts as full, unencoded, signed 64-bit integer, stored
+ *   in little-endian order;
+ *
+ * - [optional] all the other fields (nsec, tzoffset, tzindex) if any of them
+ *   were having not 0 value. They are packed naturally in little-endian order;
+ *
+ */
+
+package datetime
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"time"
+
+	"encoding/binary"
+
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+// Datetime external type
+// Supported since Tarantool 2.10. See more details in issue
+// https://github.com/tarantool/tarantool/issues/5946
+const Datetime_extId = 4
+
+/**
+ * datetime structure keeps number of seconds and
+ * nanoseconds since Unix Epoch.
+ * Time is normalized by UTC, so time-zone offset
+ * is informative only.
+ */
+type datetime struct {
+	// Seconds since Epoch
+	seconds int64
+	// Nanoseconds, fractional part of seconds
+	nsec int32
+	// Timezone offset in minutes from UTC
+	// (not implemented in Tarantool, https://github.com/tarantool/tarantool/issues/6751)
+	tzOffset int16
+	// Olson timezone id
+	// (not implemented in Tarantool, https://github.com/tarantool/tarantool/issues/6751)
+	tzIndex int16
+}
+
+const (
+	secondsSize  = 8
+	nsecSize     = 4
+	tzIndexSize  = 2
+	tzOffsetSize = 2
+)
+
+func encodeDatetime(e *msgpack.Encoder, v reflect.Value) error {
+	var dt datetime
+
+	tm := v.Interface().(time.Time)
+	dt.seconds = tm.Unix()
+	nsec := tm.Nanosecond()
+	dt.nsec = int32(math.Round((10000 * float64(nsec)) / 10000))
```

Comment:
Nano is 10^-9, I don't get what is going on here

Reply:
Fixed.